cfpb/consumerfinance.gov
# Add `ingress-nginx` to local deployments

Instead of exposing ports, it is more "prod-like" to access it through an ingress. This PR adds instructions to [`helm/README.md`](helm/README.md) explaining how to install `ingress-nginx`, and how to access the
deployment through the ingress locally.

## Additions

* Adds notes on how to install `ingress-nginx`
* Changes `helm-install.sh` to use `RELEASE` variable for ingress hostname.
  * default is `cfgov.localhost`, otherwise `<release>.localhost`.

## Changes

- default `LoadBalancer` on port `8000`, changed to `ClusterIP` for default. 


## How to test this PR

0. Make sure you have 8 CPUs and 8GB of RAM allocated in Docker Desktop Settings -> Resources.
1. Install `ingress-nginx` following the instructions.
2. Run `./build-images.sh && ./helm-install.sh`
3. Run `RELEASE=my-test ./helm-install.sh`
4. Scale `my-test-cfgov` Deployment to `0` and load different data (ex, `test.sql.gz`, follow `Manually Load Data` in `README.md`). Scale `my-test-cfgov` back to 1 once data is loaded.
5. Access [`cfgov.localhost`](http://cfgov.localhost) and [`my-test.localhost`](http://my-test.localhost) 


## Screenshots
![image](https://user-images.githubusercontent.com/59629491/182320428-1f1a8c8f-d00b-46a6-b00a-b395d4e57ec7.png)
![image](https://user-images.githubusercontent.com/59629491/182320113-256af822-4169-4553-b946-12e270f64c82.png)
